src/python/pants/backend/java/dependency_inference/java_parser_test.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.java.compile.javac import rules as javac_rules
+from pants.backend.java.compile.javac_binary import rules as javac_binary_rules
+from pants.backend.java.dependency_inference.java_parser import (
+    FallibleJavaSourceDependencyAnalysisResult,
+)
+from pants.backend.java.dependency_inference.java_parser import rules as java_parser_rules
+from pants.backend.java.dependency_inference.java_parser_launcher import (
+    rules as java_parser_launcher_rules,
+)
+from pants.backend.java.dependency_inference.types import JavaImport, JavaSourceDependencyAnalysis
+from pants.backend.java.target_types import JavaLibrary, JavaSources
+from pants.build_graph.address import Address
+from pants.core.util_rules import source_files
+from pants.core.util_rules.external_tool import rules as external_tool_rules
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.internals.scheduler import ExecutionError
+from pants.engine.process import ProcessExecutionFailure
+from pants.engine.target import Sources
+from pants.jvm.resolve.coursier_fetch import CoursierResolvedLockfile
+from pants.jvm.resolve.coursier_fetch import rules as coursier_fetch_rules
+from pants.jvm.resolve.coursier_setup import rules as coursier_setup_rules
+from pants.jvm.target_types import JvmDependencyLockfile
+from pants.jvm.util_rules import rules as util_rules
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        preserve_tmpdirs=True,
+        rules=[
+            *coursier_fetch_rules(),
+            *coursier_setup_rules(),
+            *external_tool_rules(),
+            *java_parser_launcher_rules(),
+            *java_parser_rules(),
+            *javac_binary_rules(),
+            *javac_rules(),
+            *source_files.rules(),
+            *util_rules(),
+            QueryRule(FallibleJavaSourceDependencyAnalysisResult, (SourceFiles,)),
+            QueryRule(JavaSourceDependencyAnalysis, (SourceFiles,)),
+            QueryRule(SourceFiles, (SourceFilesRequest,)),
+        ],
+        target_types=[JvmDependencyLockfile, JavaLibrary],
+        bootstrap_args=["--javac-jdk=system"],  # TODO(#12293): use a fixed JDK version.
+    )
+
+
+def test_simple_java_parser_analysis(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "coursier_resolve.lockfile": CoursierResolvedLockfile(entries=())
+            .to_json()
+            .decode("utf-8"),
+            "BUILD": dedent(
+                """\
+                coursier_lockfile(
+                    name = 'lockfile',
+                    maven_requirements = [],
```

Comment:
API change: This is now called `requirements` after my PR from last week.

Reply:
Really weird that these didn't cause any test failures.  Done.